elfo-core/src/restarting/config.rs

```diff
@@ -0,0 +1,48 @@
+use std::{num::NonZeroU64, time::Duration};
+
+use serde::Deserialize;
+
+use crate::restarting::restart_policy::{RestartParams, RestartPolicy};
+
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct RestartPolicyConfig(Option<WhenConfig>);
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "when")]
+enum WhenConfig {
+    Always(RestartParamsConfig),
+    OnFailure(RestartParamsConfig),
+    Never,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct RestartParamsConfig {
+    #[serde(with = "humantime_serde")]
+    min_backoff: Duration,
+    #[serde(with = "humantime_serde")]
+    max_backoff: Duration,
+    #[serde(with = "humantime_serde", default)]
+    auto_reset: Option<Duration>,
+    max_retries: Option<NonZeroU64>,
+    #[serde(default)]
+    factor: Option<f64>,
```

Comment:
If defined, it MUST be `> 0.0`

Reply:
will emit warning and default to safe value to not panic at runtime.